test/performance/performance.go

```diff
@@ -0,0 +1,333 @@
+		)
+	}
+
+	table.Flush()
+}
+
+// export table of results to a csv file
+func exportTable(data []metricData, filename string) {
+	f, err := os.Create(filename)
+	if err != nil {
+		klog.Exitf("Error: failed to create %s; %s", filename, err)
+	}
+
+	defer func() {
+		if cerr := f.Close(); cerr != nil && err == nil {
+			err = cerr
+		}
+	}()
+
+	w := csv.NewWriter(f)
+	defer w.Flush()
+
+	line := []string{"time", "num_policies", "avg_cpu_controller", "max_cpu_controller", "avg_cpu_apiserver", "max_cpu_apiserver"}
+	if err := w.Write(line); err != nil {
+		klog.Exitf("Error writing headers to file; %s", err)
+	}
+
+	for _, entry := range data {
+		line = []string{
+			entry.timestamp,
+			fmt.Sprintf("%d", entry.numPolicies),
+			fmt.Sprintf("%.5f", entry.controllerAvg),
+			fmt.Sprintf("%.5f", entry.controllerMax),
+			fmt.Sprintf("%.5f", entry.apiserverAvg),
+			fmt.Sprintf("%.5f", entry.apiserverMax),
+		}
+		if err := w.Write(line); err != nil {
+			klog.Exitf("Error writing data to file; %s", err)
+		}
+	}
+}
+
+func init() {
+	klog.SetOutput(os.Stdout)
+	klog.InitFlags(flag.CommandLine)
+}
+
+func main() {
+	// pull in test variables from flags
+	pflag.CommandLine.AddGoFlagSet(flag.CommandLine)
+
+	var plc_filename, output_filename string
+	var n_per_batch, n_total, per_batch_sleep int
+
+	pflag.StringVarP(&plc_filename, "policy", "p",
+		performance_dir+"/resources/templates/deployment-plc.yaml", "path to test policy YAML, scoped to performance directory")
+	pflag.IntVarP(&n_per_batch, "policies-per-batch", "b", 100, "number of policies to create per batch")
+	pflag.IntVarP(&n_total, "total-policies", "t", 1000, "total number of policies created")
+	pflag.IntVarP(&per_batch_sleep, "sleep", "s", 20, "time (min) to sleep after creating a batch of policies")
+	pflag.StringVar(&output_filename, "csv", performance_dir+"results.csv", "path to CSV to export results to")
+
+	pflag.Parse()
+
+	// wait for config map to be applied properly before executing command
+	_, err := bash("oc get secret -n openshift-user-workload-monitoring | grep  prometheus-user-workload-token")
+	if err != nil {
+		klog.V(2).Info("Setting up secret for k8s metrics")
+		_, err = exec.Command("kubectl", "apply", "-f", performance_dir+"/resources/setup/metrics-configmap.yaml").CombinedOutput()
+		if err != nil {
+			klog.Exitf("Error applying metrics configMap: %s", err)
+		}
+		// wait for secret to be created for 2 minutes
+		found := false
+		for i := 0; i < 120; i += 5 {
+			_, err = bash("oc get secret -n openshift-user-workload-monitoring | grep  prometheus-user-workload-token")
+			if err == nil {
+				found = true
+				break
+			}
+			klog.V(2).Info("Secret not found, waiting 5 seconds...")
+			time.Sleep(5 * time.Second)
+		}
+
+		if !found {
+			klog.Fatal("Error: Prometheus secret creation timed out")
+		}
+	}
+
+	secret, err := bash("oc get secret -n openshift-user-workload-monitoring | grep  prometheus-user-workload-token | head -n 1 | awk '{print $1 }'")
```

Comment:
Could you please convert this to Go for parsing the strings?

Optionally, it'd be cleaner if you used the `client-go` library so you don't have to do string parsing.